kolibri/plugins/registry.py

```diff
@@ -49,8 +49,8 @@ def initialize():
                 plugin_module = importlib.import_module(app + ".kolibri_plugin")
                 logger.debug("Loaded kolibri plugin: {}".format(app))
                 plugin_classes = []
-                for obj in plugin_module.__dict__.values():
-                    if type(obj) == type and obj is not KolibriPluginBase and issubclass(obj, KolibriPluginBase):
+                for obj in plugin_module.PLUGINS:
```

Comment:
Wonderful! This one solves it, disregard the __all__ thing, I thought that
would solve something but I was utterly and dreadfully wrong...
http://stackoverflow.com/a/21563930/405682

On 11 February 2016 at 17:40, Richard Tibbles <notifications@github.com>
wrote:

> In kolibri/plugins/registry.py
> <https://github.com/learningequality/kolibri/pull/16#discussion_r52628166>
> :
>
> > @@ -49,8 +49,8 @@ def initialize():
> >                  plugin_module = importlib.import_module(app + ".kolibri_plugin")
> >                  logger.debug("Loaded kolibri plugin: {}".format(app))
> >                  plugin_classes = []
> > -                for obj in plugin_module.__dict__.values():
> > -                    if type(obj) == type and obj is not KolibriPluginBase and issubclass(obj, KolibriPluginBase):
> > +                for obj in plugin_module.PLUGINS:
>
> With __all__ could we then just iterate over the module itself?
> … <#-1274806653_>
> On Thu, 11 Feb 2016, 02:28 Benjamin Bach <notifications@github.com>
> wrote: In kolibri/plugins/registry.py <#16 (comment)
> <https://github.com/learningequality/kolibri/pull/16#discussion_r52583577>>
> : > @@ -49,8 +49,8 @@ def initialize(): > plugin_module =
> importlib.import_module(app + ".kolibri_plugin") > logger.debug("Loaded
> kolibri plugin: {}".format(app)) > plugin_classes = [] > - for obj in
> plugin_module.__dict__.values(): > - if type(obj) == type and obj is not
> KolibriPluginBase and issubclass(obj, KolibriPluginBase): > + for obj in
> plugin_module.PLUGINS: Aaaaah, that was the thing: Could you add an __all__
> instead to the module? — Reply to this email directly or view it on GitHub <
> https://github.com/learningequality/kolibri/pull/16/files#r52583577>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/kolibri/pull/16/files#r52628166>.
>